### PR TITLE
doxygen: 1.8.19 -> 1.8.20

### PR DESCRIPTION
**Pull request by ajs124**
###### Motivation for this change
Replace #100881

That was against master, plus I took the chance to make some more changes.
1. move doxygen_gui to qt5
2. modernize the expression a bit
3. disable enableParallel, since it reproducably failed on my dual-socket Xeon machine

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

cc @NeQuissimus

### CODE REVIEW

File: pkgs/development/tools/documentation/doxygen/default.nix [hunk 1]
```diff
@@ -1,15 +1,14 @@
...
-    ];
-    sha256 = "1lvqfw2yzba588c5ggl8yhw7aw4xkk44mrghsd9yqlajc48x25dc";
+  pname = "doxygen";
+  version = "1.8.20";
+
+  src = fetchFromGitHub {
+    owner = "doxygen";
+    repo = "doxygen";
+    rev = "Release_${stdenv.lib.replaceStrings [ "." ] [ "_" ] version}";
+    sha256 = "17chvi3i80rj4750smpizf562xjzd2xcv5rfyh997pyvc1zbq5rh";
```
Comment (vcunat) [inline pkgs/development/tools/documentation/doxygen/default.nix:11]: GitHub is preferred to upstream tarballs?

Comment (ajs124) [reply pkgs/development/tools/documentation/doxygen/default.nix:11]: Not sure if we have any policy on that kind of thing, but personally, I prefer downloading tags from a git repository vs. tarballs from random project sites. Plus, that way update bots like @r-ryantm can potentially pick up new releases.

Comment (ryantm) [reply pkgs/development/tools/documentation/doxygen/default.nix:11]: The bot doesn't care either way for updates. You can see there are many bot updates of this package in the history: https://github.com/NixOS/nixpkgs/commits/1f75b60c56688d79366ceb7cd11b597fc680dbfd/pkgs/development/tools/documentation/doxygen/default.nix

Comment (ajs124) [reply pkgs/development/tools/documentation/doxygen/default.nix:11]: Right, nevermind that part then. I can also revert to fetchurl, if that's preferred, although I can see arguments for and against both ways.

Comment (ryantm) [reply pkgs/development/tools/documentation/doxygen/default.nix:11]: Personally, I trust GitHub more than sourceforge and the canonical file mirror isn't https, so I like this change.

Comment (vcunat) [reply pkgs/development/tools/documentation/doxygen/default.nix:11]: OK.

I managed to get https from the canonical URL, though maybe there's something weird about it as my `nix-prefetch-url` wouldn't stomach it.

**Review by NeQuissimus — APPROVED** [review]

**Pull request merged**